keda/templates/12-keda-deployment.yaml

```diff
@@ -40,11 +40,11 @@ spec:
       {{- end }}
       serviceAccountName: {{ .Values.serviceAccount.name }}
       securityContext:
-        {{- toYaml .Values.podSecurityContext | nindent 8 }}
+        {{- toYaml .Values.podSecurityContext.operator | nindent 8 }}
```

Comment:
Can you make it backwards compatible please? So that it uses .Values.podSecurityContext when available, otherwise .Values.podSecurityContext.operator when specified

Reply:
+1

Reply:
Sure, updated. Please check again.

Reply:
The solution looks nice but I think is not enough, What about not exiting `.Values.podSecurityContext.operator` but existing `.Values.podSecurityContext.metricServer`

Reply:
THat's true, however, we use an empty object by default so think that should be fine (but to be verified)

Reply:
May be we can do something like,

![image](https://user-images.githubusercontent.com/14357598/151356378-77419785-bd76-4ebe-97ac-ff79bdea0f09.png)

But I am not sure if we are over-complicating the things by making this backward compatible ?

Reply:
Ideally we could have made 2 top level sections for operator and metricServer and put parameters specific to those under respective section and common parameters off course live at top level, something like -

![image](https://user-images.githubusercontent.com/14357598/151359212-b7b0b0ab-ef5a-4d4f-8155-a19171e2ecb3.png)

But I guess that's out of scope for this PR :)

Reply:
That is a fully rework and should be with the new major version imo

Reply:
Right. I don't want to break existing stuff, lol. But that's more of a suggestion for overall refactoring in future. But what about https://github.com/kedacore/charts/pull/239#discussion_r793545246 ?

Reply:
I don't think we need that, but thanks for asking!

Reply:
Ok cool

Reply:
In fact, [we discussed](https://github.com/kedacore/charts/issues/182) it and we postponed till next major release.

For the moment, I'd maintain the backward compatibility and keep the refactors for the future. Maybe achieving this is more complicated from the chart pov, but IMHO it's the best approach.

Reply:
> The solution looks nice but I think is not enough, What about not exiting `.Values.podSecurityContext.operator` but existing `.Values.podSecurityContext.metricServer`?

Even then we should use the current approach IMO because it's not because they define something for the metric server that the operator should have something

Reply:
So just confirming if we are good with what we have in the end ?

Reply:
> > The solution looks nice but I think is not enough, What about not exiting `.Values.podSecurityContext.operator` but existing `.Values.podSecurityContext.metricServer`?
> 
> Even then we should use the current approach IMO because it's not because they define something for the metric server that the operator should have something

@JorTurFer Thoughts on this? Because with your current approach you enforce people to provide operator info when metric server info is specified.

Reply:
We shouldn't assume that there is not a special user with super edge case, but for me it's enough if we continue as it is right now, adding a note explaining the limitation. Probably I'm overthinking the things, if we have any use case, we can do it and release a new chart version

Reply:
> We shouldn't assume that there is not a special user with super edge case

But what is the super edge case? I'm not sure I'm getting the scenario you are concerned about?

What I believe we should provide is (similar for resources):
- `podSecurityContext.metricServer`: Applies pod security context to metrics server
- `podSecurityContext.operator`: Applies pod security context to operator
- `podSecurityContext`: Applied to both

Container specific variable takes priority over `podSecurityContext`. If they specify `podSecurityContext.operator` but not `podSecurityContext.metricServer` then that's their concern, we should not block them because of it im.

Reply:
What are you thoughts @zroubalik ?

Reply:
> Container specific variable takes priority over podSecurityContext. If they specify podSecurityContext.operator but not podSecurityContext.metricServer then that's their concern, we should not block them because of it imo.

Yes, that's what I wanted to express :)

Reply:
Ok, in that case we can merge the PR I think - Do you agree?

Reply:
yes